nni/retiarii/execution/base.py

```diff
@@ -123,8 +136,8 @@ def query_available_resource(self) -> int:
         return self.resources
 
     def budget_exhausted(self) -> bool:
-        advisor = get_advisor()
-        return advisor.stopping
+        resp = rest.get(self.port, '/check-status', self.url_prefix)
```

Comment:
maybe we can, but it would change a lot. let's improve this part step by step.
this functionality is provided by execution engine. for now, advisor is only the channel to send trial configurations.